crates/algorithms/src/haversine/mod.rs

```diff
@@ -0,0 +1,131 @@
+use core::fmt::Debug;
+use nalgebra::Point2;
+use num_traits::Float;
+
+///calculate distance between point with use haversine formula
+/// return distance in km
+#[cfg_attr(
+    feature = "tracing",
+    tracing::instrument("Calculate Haversine Data", skip_all)
+)]
+fn calculate_haversine_distance<T>(p1: Point2<T>, p2: Point2<T>, sphere_radius: T) -> T
+where
+    T: Copy + Float + Debug + 'static,
```

Comment:
I believe this can be surmised as "T: Scalar + Float," as Scalar is the trait required by nalgebra::OPoint<T, 2>, and Float is the minimal trait which gives us the to_radians and to_degrees functions :)